server/src/servershot.js

```diff
@@ -100,12 +100,16 @@ class Shot extends AbstractShot {
         // If the key doesn't already exist, go through the clips being inserted and
         // check to see if we need to store any data: url encoded images
         return this.convertAnyDataUrls(client, json, possibleClipsToInsert).then((oks) => {
+          let head = json.head;
+          let body = json.body;
+          json.head = null;
+          json.body = null;
           return db.queryWithClient(
             client,
-            `INSERT INTO data (id, deviceid, value)
-             VALUES ($1, $2, $3)`,
-            [this.id, this.ownerId, JSON.stringify(json)]
-          ).then((rows) => {
+            `INSERT INTO data (id, deviceid, value, head, body)
+             VALUES ($1, $2, $3, $4, $5)`,
+            [this.id, this.ownerId, JSON.stringify(json), head, body]
+          ).then((rowCount) => {
```

Comment:
Should we be updating the client and telling it not to keep sending the head and body (similar to what we do with clips)?  And then we'd only overwrite head and body if the json head/body weren't null?